### ISSUE TITLE
Form Submissions Export - Nicer Filename

### ISSUE THREAD

**Issue by lb-** [wagtail/wagtail#3812]
I think it might be nicer to have a better filename on CSV exports of Form Submissions.

The following lines would be replaced as follows:
https://github.com/wagtail/wagtail/blob/master/wagtail/wagtailforms/views.py#L89

This would add the slug string of the page to the filename:

```python
    if request.GET.get('action') == 'CSV':
        filename = '%s-export.csv' % (form_page.slug)
        # return a CSV instead
        response = HttpResponse(content_type='text/csv; charset=utf-8')
        response['Content-Disposition'] = 'attachment;filename=%s' % (filename)
```

We could potentially add an ISO date to the filename also.

Happy to do this, just want to check if it is going to help.

**Comment by thibaudcolas** [comment]
That's pretty much what I'm adding manually every.single.time I work with this form export, including the ISO date (well, just YYYYMMDD). Would definitely help me at least!

**Comment by lb-** [comment]
@thibaudcolas 

Maybe we should have a format:

`<form-page-slug>-export-<YYYY-MM-DD>.csv`

Where the date is the current date, ISO format.

**Comment by lb-** [comment]
This is fixed in the PR #3866 with a method `get_csv_filename` that will return the filename with the ISO date in it, looks like it can be overridden easily by developers also.

When that PR closes, this can close also.

**Comment by lb-** [comment]
Implemented in #4107 along with docs on how to easily customise the csv export filename.
